### PR TITLE
Fix a possible memleak in eckey_priv_encode

### PR DESCRIPTION
**Pull request by bernd-edlinger**
Additionally use OPENSSL_clear_free on the private key data in case of error.

master has the memory leak already fixed, just the clear free on the private key is missing.

### CODE REVIEW

**Review by t8m — APPROVED** [review]
OK, if you promise to create the patch for master.

**Comment by t8m** [comment]
It needs review from committers ASAP if it should get into 1.1.1

**Review by hlandau — APPROVED** [review]

**Comment by bernd-edlinger** [comment]
Merged to 1.1.1 branch.  Thanks!

**Pull request closed without merging**
